tests/test_server.py

```diff
@@ -245,6 +246,28 @@ def test_get_all_shopcart(self):
         carts = json.loads(resp.data.decode('utf8'))
         self.assertEqual( len(carts), 3)
 
+    def test_query_product(self):
+        cart = server.Shopcart.find(1)
+        cart.add_product( 5 )
+        cart.add_product( 6 )
+        cart.save() 
+        cart = server.Shopcart.find(2)
+        cart.add_product( 6 )
+        cart.save()
+        self.assertEqual(len(server.Shopcart.find_by_product(123)), 0)
```

Comment:
Probably don't need this assertion as this should be handled in the `test_shopcart.py` file.

Reply:
Now I never know what's in memory since we have so many test so I'm just getting sure it's not there. I think it should be there because maybe in the future we add some test data in the middle for soooome reason and it would be a pain to be sure. It's not so much of a test  that this command works becase it's already been tested in the model.

Reply:
Yeah, that's why I said I was being picky. I found myself sometime putting that but had to remind myself it wasn't necessary.

All these comments were just personal preference. I'm going to approve the PR. Feel free to change any or none of the things I said.